### ISSUE TITLE
prefect server start failes to start graphql container on 0.14.10 (but 0.13.19 works)

### ISSUE THREAD

**Issue by Spooky-0** [PrefectHQ/prefect#4170]
## Description
<!-- A clear description of the bug -->

I am a first time user of prefect - switching over from airflow.

Just downloaded the latest version 0.14.10

I tried running 
```~/Documents/preked$ prefect server start                                                                                                                                          [512/564]
Pulling postgres ... done                                                                                                                                                                                  
Pulling hasura   ... done                                                                                                                                                                                  
Pulling graphql  ... done                                                                                                                                                                                  
Pulling apollo   ... done                                                                                                                                                                                  
Pulling towel    ... done                                                                                                                                                                                  
Pulling ui       ... done                                                                                                                                                                                  
Building with native build. Learn about native build in Compose here: https://docs.docker.com/go/compose-native-build/                                                                                     
Creating network "prefect-server" with the default driver                                                                                                                                                  
Creating tmp_postgres_1 ... done                                                                                                                                                                           
Creating tmp_hasura_1   ... done                                                                                                                                                                           
                                                                                                                                                                                                           
ERROR: for graphql  Container "9c870d315bbc" is unhealthy.  
```

With this error. I retried a bunch with same error.

After switching to version 13.19 it worked immediately. 

## Expected Behavior
<!-- What did you expect to happen instead? -->

I expected the graphql container to start successfully.


## Reproduction
<!-- A minimal example that exhibits the behavior. -->


```
conda create -n testenv python=3.8
pip install prefect
prefect backend server
prefect server start
```

<!-- Any additional information about your environment
Optionally run `prefect diagnostics` from the command line and paste the information here. -->
Since I already changed to 0.13.19 I can't really run 'prefect diagnsostics' anymore...
I used a clean conda env with only Kedro and Prefect installed (via pip) on Ubuntu 20.04.

**Comment by madkinsz** [comment]
I've gotten this error spuriously since the healthchecks were added in #4041 but it's never happened repeatedly after retry. I wonder if we need to tweak the healthcheck numbers, not sure why docker-compose stops trying here.

**Comment by Spooky-0** [comment]
Sorry, as I said I am very new to Prefect so I really can't say what is going on.

**Comment by madkinsz** [comment]
> Sorry, as I said I am very new to Prefect so I really can't say what is going on.

Sorry @Spooky-0 -- that question was more so directed to other developers on the team. If that message takes 60s to display, that'd be helpful to know, but I don't expect you to dig into this. We'll look into it further 👍

**Comment by madkinsz** [comment]
@Spooky-0 could you provide your

`docker version`
`docker-compose --version`

and how much memory you have allocated to your Docker installation?

**Comment by UPetit** [comment]
@Spooky-0 I got the same issue yesterday by trying to override some parameters in my `config.toml` file.
For instance, I think overriding password db could raise an issue if the password has already been set previously.
In my case, I removed everything I overrided that were not necessary and it became healthy again. Hope it helps

**Comment by Spooky-0** [comment]
@madkinsz 
```
(base) j@jlap:~$ docker version
docker-Client: Docker Engine - Community
 Version:           20.10.3
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        48d30b5
 Built:             Fri Jan 29 14:33:21 2021
 OS/Arch:           linux/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.3
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       46229ca
  Built:            Fri Jan 29 14:31:32 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
(base) j@jlap:~$ docker-compose --version
docker-compose version 1.28.4, build cabd5cfb
(base) j@jlap:~$ 
```
I'm not sure how to check how much memory I have allocated to Docker Engine... My laptlop only has 8gb ram and I am running Ubuntu nativley.

I read (some of) https://docs.docker.com/config/containers/resource_constraints/ but can't see how to answer your question regarding memeory allocation. I never set anything explicitly for docker (just defaults) and then only ran `prefect server start` and let docker-compose do its magic.

Hope this helps.

**Comment by madkinsz** [comment]
Thanks @Spooky-0 ! We're having a hard time reproducing but docker versions are complicated so this is helpful for narrowing it down. Perhaps resource allocation is just a Docker Desktop thing (https://docs.docker.com/docker-for-mac/#resources).

**Comment by brizzbane** [comment]
I just started getting this error, and I successfully ran before. (wiped machine and started new project--now it doesn't run)

**Comment by willianchan** [comment]
having the same error here with version 0.14.11

**Comment by madkinsz** [comment]
Hi @brizzbane and @willianchan can you include the output of `docker version` and `docker-compose --version` as well please?

**Comment by willianchan** [comment]
Nevermind, i did a reset to factory defaults at docker and the error disappeared.

My docker version:

```
Client: Docker Engine - Community
 Cloud integration: 1.0.4
 Version:           20.10.0
 API version:       1.41
 Go version:        go1.13.15
 Git commit:        7287ab3
 Built:             Tue Dec  8 18:55:31 2020
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server: Docker Engine - Community
 Engine:
  Version:          20.10.0
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       eeddea2
  Built:            Tue Dec  8 18:58:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.4.3
  GitCommit:        269548fa27e0089a8b8278fc4fc781d7f65a939b
 runc:
  Version:          1.0.0-rc92
  GitCommit:        ff819c7e9184c13b7c2607fe6c30ae19403a7aff
 docker-init:
  Version:          0.19.0
  GitCommit:        de40ad0
```

My docker compose version:
```
docker-compose version 1.27.4, build 40524192
docker-py version: 4.3.1
CPython version: 3.7.4
OpenSSL version: OpenSSL 1.1.1c  28 May 2019
```

**Comment by willianchan** [comment]
any resolution updates? I restarted the computer and the error occurred again, I can't get it to work...

**Comment by madkinsz** [comment]
I'd recommend setting a higher memory limit. I'm glad it was as simple as resetting your config though!

**Comment by mastoltejr** [comment]
Hello, I'm having this same issue as well on Windows.

I've reset docker desktop to factory defaults and have allocated more memory. Neither has worked for me.

**Comment by mazar** [comment]
Due to an issue with 0.14.12, I tried to downgrade to 0.14.11 (and 0.14.10) and got to the same issue.
`ERROR: for graphql  Container "abcdef" is unhealthy. `
Going back to `0.14.12` fixes this issue (but 0.14.12 has its own problems)
